File: c#/state-template/src/dominio/estados/EstadoConcretoB.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace state_template.src.dominio.estados
{
    class EstadoConcretoB : EstadoContexto 
    {
        public void actuar(Contexto contexto) 
        {
            //TODO
            if (cumpleCondicionParaCambiarDeEstado(contexto))
            {
                //cambiar a otro estado
            }
        }

        private bool cumpleCondicionParaCambiarDeEstado(Contexto contexto)
        {
            //TODO
            return true;
        }
    }
}

File: python/src/estados/estado_concreto_b.py
from src.estados.estado_contexto import EstadoContexto


class EstadoConcretoB(EstadoContexto):

    def actuar(self, contexto):
        # TODO
        if self.__cumple_condicion_para_cambiar_de_estado(contexto):
            # cambiar de estado
            pass

    def __cumple_condicion_para_cambiar_de_estado(self, contexto):
        # TODO
        return True
